src/pages/index.js

```diff
@@ -1,256 +1,231 @@
-
-// DELETE
-const deleteConfirmModal = new PopupWithConfirm(
-  "#delete-check-modal",
-  handleCardDeleteSubmit
-);
-deleteConfirmModal.setEventListeners();
-
-// AVATAR
-const avatarChangeModal = new PopupWithForm(
-  "#avatar-change-modal",
-  handleAvatarChangeSubmit
-);
-avatarChangeModal.setEventListeners();
-
-///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-
-//USER INFO
-const user = new UserInfo(
-  ".profile__name",
-  ".profile__description",
-  ".profile__picture"
-);
-
-///////////////////////////////////////////  FUNCTION   /////////////////////////////////////////////////////////////////////////////
+  // PROFILE INFO
+const userInfo = new UserInfo({
+  userName: "#profile-title-js",
+  userAbout: "#profile-description-js",
+  userAvatar: ".profile__image",
+});
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+  })
+  .catch((err) => {
+    console.log("Failed to fetch user info:", err);
+  });
 
-// IMAGE CLICK
-function handleImageClick(title, link) {
-  previewImageModal.open(title, link);
-}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-// CREATE CARD
-function makeCard(cardData) {
+// CARD DATA
+function getCardElement(cardData) {
   const card = new Card(
     cardData,
-    cardTemplate,
+    cardSelector,
     handleImageClick,
-    handleDeleteClick,
-    handleCardLike,
-    checkIfLiked
+    handleCardDelete,
+    handleLikeClick
   );
-  return card.generateCard();
-}
-
-// LIKE
-function checkIfLiked(ifLiked, element) {
-  if (ifLiked === true) {
-    element
-      .querySelector(".heart-button")
-      .classList.toggle("heart-button_active");
-  }
+  return card.getView();
 }
 
-// DELETE
-function handleDeleteClick(cardId) {
-  console.log(cardId);
-  deleteConfirmModal.open(cardId);
-}
-
-/////////////////////////////////////////////////  VALIDATOR  ////////////////////////////////////////////////////////////////////////////////////
-
-// PROFILE
-const editFormValidator = new FormValidator(config, profileEditForm);
-
-// CARD
-const cardFormValidator = new FormValidator(config, cardAddForm);
-
-// AVATAR
-const avatarFormValidator = new FormValidator(config, avatarEditForm);
+//////////////////////////////////////////////////////////////  FORM  //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-avatarFormValidator.enableValidation();
-editFormValidator.enableValidation();
-cardFormValidator.enableValidation();
+// EDIT PROFILE
+const profileEditPopup = new PopupWithForm(
+  "#profile-edit-modal",
+  (formData) => {
+    profileEditPopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```